python/tank/descriptor/bundle_cache_usage/manager.py

```diff
@@ -0,0 +1,270 @@
+        trying to delete a file that exists in a symlinked folder. Therefore, we'll first
+        scan the entire list for a link, symlink or such in the list. If found, we'll
+        abort the deletion process before actually deleting anything.
+
+        :param filelist: A list file and folders to be deleted
+        """
+
+        # First, check whether there is a symlink in the list
+        for f in filelist:
+            if os.path.islink(f):
+                # CAVEAT: Always False if symbolic links are not supported by the Python runtime.
+                #         How do we know whether it is supported???
+                raise BundleCacheUsageFileDeletionException(f, "Found a symlink")
+
+        # We have a crude list, now we need to sort it out in reverse
+        # order so we can later on delete files, and then parent folder
+        # in a logical order.
+        rlist = list(reversed(filelist))
+        # No symlinks, Houston we're clear for deletion
+        for f in rlist:
+            if not os.path.exists(f):
+                raise BundleCacheUsageFileDeletionException(
+                    f,
+                    "Attempting to delete non existing file or folder."
+                )
+
+            if os.path.isfile(f):
+                os.remove(f)
+
+            elif os.path.isdir(f):
+                # Because we're deleting items that should be reverse prdered
+                # when we're about to delete a folder, it should be empty already.
+                # let's check it out!
+                try:
+                    os.rmdir(f)
+                except OSError as e:
+                    raise BundleCacheUsageFileDeletionException(
+                        f,
+                        "Attempted to delete a non-empty folder: %s" % (e)
+                    )
+
+            else:
+                raise BundleCacheUsageFileDeletionException(
+                    f,
+                    "Not a link, not a file, not a directory ???"
+                )
+
+    ###################################################################################################################
+    #
+    # PUBLIC API - methods & properties
+    #
+    # Can be called from any threading contextes (main or worker)
+    #
+    ###################################################################################################################
+
+    @property
+    def bundle_cache_root(self):
+        return self._bundle_cache_root
+
+    def initial_populate(self):
+        """
+        Scan the bundle cache at the specified location for bundles and add them
+        as unused entries to the database.
+
+        """
+
+        log.info("Searching for existing bundles ...")
+        start_time = time.time()
+        found_bundles = self._find_bundles()
+        for bundle_path in found_bundles:
+            self._worker.add_unused_bundle(bundle_path)
+
+        log.info("populating done in %ss, found %d entries" % (
```

Comment:
you can use the http://developer.shotgunsoftware.com/tk-core/utils.html#sgtk.log.LogManager.log_timing decorator!